### PR TITLE
Add a docker file for building a container image

### PR DESCRIPTION
**Pull request by portante**
And update the `package-lock.json` file.

### CODE REVIEW

File: containers/Dockerfile
Comment (sjmonson) [inline containers/Dockerfile]: Rather then depending on the host's npm why don't we just do everything in containers?
An unoptimized build stage could be something like:
```Dockerfile
FROM registry.fedoraproject.org/fedora-minimal:37 AS BUILD
USER root
RUN microdnf update -y && microdnf install -y nodejs
ADD . /sc
WORKDIR /sc
RUN npm install && npm run build
```

Followed by copying the files into the final container:
```Dockerfile
COPY --from=BUILD /var/cache/yum /var/cache/yum
COPY --from=BUILD /sc/dist/src /smee-client
COPY --from=BUILD /sc/package.json /smee-client/
```

**Review by sjmonson — APPROVED** [review]

File: containers/Dockerfile
Comment (portante) [reply containers/Dockerfile]: @scrufulufugus, let me know what you think.

**Review by portante — COMMENTED** [review]
Running in our Jenkins environment successfully so far.

**Review by sjmonson — APPROVED** [review]
Looks good to me. I don't have write access to the repo so my approval does not count towards the limit.

**Review by webbnh — APPROVED** [review]
👍

**Comment by portante** [comment]
> I don't have write access to the repo so my approval does not count towards the limit.

Once you accept the invite to become a `smee-client-core` team member you will have admin access.

**Pull request merged**
